src/js/api/set-config.js

```diff
@@ -0,0 +1,71 @@
+define([
+    'utils/underscore',
+], function(_) {
+
+    const supportedFields = [
+        'repeat',
+        'volume',
+        'mute',
+        'autostart'
+    ];
+    let model;
+
+    function setVolume(mute, volume) {
+        const muteSet = !_.isUndefined(mute);
+        const volumeSet = !_.isUndefined(volume);
+
+        if (volumeSet) {
+            model.setVolume(volume);
+            if (!muteSet) {
+                model.setMute(false);
+            }
+        }
+
+        if (muteSet) {
+            model.setMute(mute);
+        }
+    }
+
+    function setAutostart(controller, autoStart) {
+        model.set('autostart', autoStart);
+
+        const autoStartOnMobile = model.autoStartOnMobile();
+        if (autoStartOnMobile) {
+            model.set('autostartMuted', true);
+        }
+        model.set('playOnViewable', autoStartOnMobile || autoStart === 'viewable');
+
+        if (model.get('state') === 'idle' && autoStart === true) {
+            controller._play({ reason: 'autostart' });
+        }
+    }
+
+    return (controller, newConfig = {}) => {
```

Comment:
I believe it would be more testable to pass in `controller` and `model` instead of just `controller`